### PR TITLE
feat(modal): support pass mousePosition to control modal's animation origin position

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

- #38571 

### 💡 Background and solution

support pass mousePosition to control modal's animation target position

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Allow Modal pass `mousePosition` to control animation origin position. |
| 🇨🇳 Chinese | Modal 允许透传 `mousePosition` 来控制弹框动画原点。 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

### CODE REVIEW

File: components/modal/index.zh-CN.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/3StSdUlSH/Modal.svg
 | zIndex | 设置 Modal 的 `z-index` | number | 1000 |  |
 | onCancel | 点击遮罩层或右上角叉或取消按钮的回调 | function(e) | - |  |
 | onOk | 点击确定回调 | function(e) | - |  |
+| mousePosition | 接收外部传入的坐标信息用于控制弹框缩放原点 | { x: number; y: number } | null | - |  |
```
Comment (afc163) [inline components/modal/index.zh-CN.md:50]: 不作为属性透给用户了，这个不是常用功能。

File: components/modal/demo/custom-mouse-position.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+---
+order: 0
+title:
+  zh-CN: 控制弹框动画原点
+  en-US: control modal's animation origin position
```
Comment (afc163) [inline components/modal/demo/custom-mouse-position.md:5]: 放一个 debug 演示即可

File: components/modal/demo/custom-mouse-position.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    <>
+      <Button type="primary" onClick={showModal}>
+        Open Modal
+      </Button>
+      <Modal 
+        title="Basic Modal" 
+        open={isModalOpen} 
+        onOk={handleOk} 
+        onCancel={handleCancel} 
+        mousePosition={{x: 300, y: 300}}
```
Comment (afc163) [inline components/modal/demo/custom-mouse-position.md:45]:
```suggestion
        mousePosition={{ x: 300, y: 300 }}
```

File: components/modal/demo/custom-mouse-position.md (hunk 2 above)
Comment (kiner-tang) [reply components/modal/demo/custom-mouse-position.md:5]: 好的

File: components/modal/demo/custom-mouse-position.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+---
+order: 0
```
Comment (afc163) [inline components/modal/demo/custom-mouse-position.md:2]: order: 999

Comment (kiner-tang) [reply components/modal/demo/custom-mouse-position.md:2]: 好的

Comment (kiner-tang) [reply components/modal/demo/custom-mouse-position.md]: done

File: components/modal/demo/custom-mouse-position.md (hunk 2 above)
Comment (kiner-tang) [reply components/modal/demo/custom-mouse-position.md:5]: done

File: components/modal/index.zh-CN.md (hunk 1 above)
Comment (kiner-tang) [reply components/modal/index.zh-CN.md]: done

**Review by afc163 — APPROVED** [review]

**Pull request merged**